### PR TITLE
Atlas craddock

### PR DESCRIPTION
**Pull request by darya-chyzhyk**
The idea of reorganization of the atlases in order to standardize the parameters and outputs of the atlas fetcher was mentioned in issue #1489 .
I started the modification with fetch_atlas_craddock_2012.

### CODE REVIEW

File: nilearn/datasets/atlas.py [hunk 1]
```diff
@@ -89,6 +90,131 @@ def fetch_atlas_craddock_2012(data_dir=None, url=None, resume=True, verbose=1):
...
+    1914-1928. doi:10.1002/hbm.21333.
+
+    See http://www.nitrc.org/projects/cluster_roi/ for more information
+    on this parcellation.
+    """
+    if (atlas_name is None) or (number_of_regions is None):
+        warnings.simplefilter("always")
+        warnings.warn("The new parameters 'atlas_name' and 'number_of_regions' "
+                      "are added in current version and will be mandatory "
+                      "in the future releases. The new implementation save "
```
Comment (bthirion) [inline nilearn/datasets/atlas.py]: 'saves'

File: nilearn/datasets/atlas.py [hunk 2]
```diff
@@ -89,6 +90,131 @@ def fetch_atlas_craddock_2012(data_dir=None, url=None, resume=True, verbose=1):
...
+                             atlas_name, '\n'.join(atlas_items)))
+
+    data_dir = _get_dataset_dir(dataset_name, data_dir=data_dir,
+                                verbose=verbose)
+    sub_files = _fetch_files(data_dir, filenames, resume=resume,
+                             verbose=verbose)
+    atlas_name_file = dict(zip(atlas_items, sub_files))
+
+    # select indicated atlas file
+    atlas_selected = atlas_name_file[atlas_name]
```
Comment (bthirion) [inline nilearn/datasets/atlas.py]: maybe 'selected_atlas'

**Comment by bthirion** [comment]
Please add the corresponding tests.

**Comment by KamalakerDadi** [comment]
FYI: CircleCI failures which are unrelated to your PR are now fixed in master. You need to rebase .

File: nilearn/datasets/atlas.py [hunk 3]
```diff
@@ -89,6 +90,131 @@ def fetch_atlas_craddock_2012(data_dir=None, url=None, resume=True, verbose=1):
...
+    Craddock, R. Cameron, G.Andrew James, Paul E. Holtzheimer, Xiaoping P. Hu,
+    and Helen S. Mayberg. "A Whole Brain fMRI Atlas Generated via Spatially
+    Constrained Spectral Clustering". Human Brain Mapping 33, no 8 (2012):
+    1914-1928. doi:10.1002/hbm.21333.
+
+    See http://www.nitrc.org/projects/cluster_roi/ for more information
+    on this parcellation.
+    """
+    if (atlas_name is None) or (number_of_regions is None):
+        warnings.simplefilter("always")
```
Comment (jeromedockes) [inline nilearn/datasets/atlas.py]: maybe use warnings.catch_warnings context manager?

**Comment by jeromedockes** [comment]
Among things that could be made more uniform, should datasets descriptions and
atlas labels (when they exist) be given as ``bytes`` or ``str``?

**Comment by KamalakerDadi** [comment]
>Among things that could be made more uniform, should datasets descriptions and
>atlas labels (when they exist) be given as bytes or str?

Yes. +1

**Comment by kchawla-pi** [comment]
Hey @darya-chyzhyk would you be able to resume work on this in the foreseeable future?

**Comment by darya-chyzhyk** [comment]
Hey, @kchawla-pi!
It's in my todo list ;) will finish it asap

**Comment by kchawla-pi** [comment]
:-) Cool! Thanks Darya!
You should do a `git pull --log upstream master` when you start, lots of changes with potential conflicts. There are big changes in  CircleCI as well.

**Comment by darya-chyzhyk** [comment]
In the new version there are two mandatory parameters: `atlas_name` and `number_of_regions`, for instance

`atlas_name = ’random’` and `number_of_regions=10`

in order to fetch this particular atlas we download "random_all.nii.gz". 

At the moment all 5 possible files are downloaded: "scorr05_mean_all.nii.gz", "tcorr05_mean_all.nii.gz", "scorr05_2level_all.nii.gz" "tcorr05_2level_all.nii.gz", "random_all.nii.gz".

The return is the path to particular file “random_10.nii.gz”.

**Question**: After extracting "random_10.nii.gz" from "random_all.nii.gz" what files its reasonable to keep in the folder?

- (a) – all files: "scorr05_mean_all.nii.gz", "tcorr05_mean_all.nii.gz", "scorr05_2level_all.nii.gz" "tcorr05_2level_all.nii.gz", "random_all.nii.gz" and “random_10.nii.gz”
- (b) – indicated with it parent file: “random_10.nii.gz” and "random_all.nii.gz" 
- (c) – just indicated: “random_10.nii.gz”

File: nilearn/datasets/atlas.py [hunk 4]
```diff
@@ -90,11 +90,10 @@ def _fetch_atlas_craddock_2012(data_dir=None, url=None, resume=True,
...
 
-def fetch_atlas_craddock_2012(atlas_name=None, number_of_regions=None,
-                              data_dir=None,
-                              url=None,
-                              resume=True,
-                              verbose=1):
+
+def fetch_atlas_craddock_2012(atlas_name, number_of_regions, data_dir=None,
+                              url=None, resume=True, verbose=1):
+
```
Comment (kchawla-pi) [inline nilearn/datasets/atlas.py]: ONe param er line is always easier to read & cleaner.

File: nilearn/datasets/atlas.py [hunk 5]
```diff
@@ -149,13 +148,15 @@ def fetch_atlas_craddock_2012(atlas_name=None, number_of_regions=None,
     on this parcellation.
     """
     if (atlas_name is None) or (number_of_regions is None):
```
Comment (kchawla-pi) [inline nilearn/datasets/atlas.py]: The parantheses are unnecessary.

File: nilearn/datasets/atlas.py [hunk 6]
```diff
@@ -149,13 +148,15 @@ def fetch_atlas_craddock_2012(atlas_name=None, number_of_regions=None,
...
-                      "are added in current version and will be mandatory "
-                      "in the future releases. The new implementation save "
-                      "the atlas on your computer and returns the atlas path "
-                      "NOT Nifti.image.",
-                      category=DeprecationWarning)
+        with warnings.catch_warnings():
+            warnings.simplefilter("always")
+            warnings.warn("The new parameters 'atlas_name' and "
+                          "'number_of_regions' are added in current version "
+                          "and will be mandatory in the future releases. The "
```
Comment (kchawla-pi) [inline nilearn/datasets/atlas.py]: mandatory in future releases.

File: nilearn/datasets/atlas.py [hunk 7]
```diff
@@ -149,13 +148,15 @@ def fetch_atlas_craddock_2012(atlas_name=None, number_of_regions=None,
...
+                          "new implementation saves the atlas on your "
+                          "computer and returns the atlas path NOT "
+                          "Nifti.image.",
```
Comment (kchawla-pi) [inline nilearn/datasets/atlas.py]: path NOT a Nifti image.
or
path NOT Nifti images.

File: nilearn/datasets/atlas.py [hunk 8]
```diff
@@ -185,8 +186,8 @@ def fetch_atlas_craddock_2012(atlas_name=None, number_of_regions=None,
     atlas_name_file = dict(zip(atlas_items, sub_files))
 
     # select indicated atlas file
```
Comment (kchawla-pi) [inline nilearn/datasets/atlas.py]: This older comment is unnecessary noise, your code explains what's happening very nicely.

File: nilearn/datasets/tests/test_atlas.py [hunk 9]
```diff
@@ -261,22 +261,16 @@ def test_fail_fetch_atlas_harvard_oxford():
...
-        "scorr05_mean_all.nii.gz",
-        "tcorr05_mean_all.nii.gz",
-        "scorr05_2level_all.nii.gz",
-        "tcorr05_2level_all.nii.gz",
-        "random_all.nii.gz",
-    ]
     assert_equal(len(tst.mock_url_request.urls), 1)
-    for key, fn in zip(keys, filenames):
-        assert_equal(bunch[key], os.path.join(tst.tmpdir, 'craddock_2012', fn))
+    fn = target_atlas + '_' + str(target_regions) + '.nii.gz'
```
Comment (kchawla-pi) [inline nilearn/datasets/tests/test_atlas.py]: Replace `fn` by `filename`.

**Comment by kchawla-pi** [comment]
I don't know the answer to your questions, I will get the attention of someone who does.

**Comment by GaelVaroquaux** [comment]
> Question: After extracting "random_10.nii.gz" from "random_all.nii.gz" what files its reasonable to keep in the folder?

All

Also, could you choose a good default for altas_name so that the user only has to worry about the dimensionality.

**Comment by illdopejake** [comment]
Hi @darya-chyzhyk -- I'm participating in the nilearn code sprint, and I was thinking of attacking the issue raised in #1489 , #1635 , etc. I see you've gotten started on things with Craddock atlas.

Are you working on this actively? If not, would you mind if I did a PR of your PR and try to whittle away at this a bit over the next few days?

**Comment by darya-chyzhyk** [comment]
> Hi @darya-chyzhyk -- I'm participating in the nilearn code sprint, and I was thinking of attacking the issue raised in #1489 , #1635 , etc. I see you've gotten started on things with Craddock atlas.
> 
> Are you working on this actively? If not, would you mind if I did a PR of your PR and try to whittle away at this a bit over the next few days?

Hi @illdopejake !
That's super cool that you have NiLearn code sprint, have fun!

#1489 issue is related to several atlases that we have in mind to improve and you can work on. The Craddock atlas is mostly done, I have to fix some small details and will finish PR very soon.

**Comment by illdopejake** [comment]
> Hi @illdopejake !
> That's super cool that you have NiLearn code sprint, have fun!
> 
> #1489 issue is related to several atlases that we have in mind to improve and you can work on. The Craddock atlas is mostly done, I have to fix some small details and will finish PR very soon.

Awesome! Thanks for the quick reply! I'll start working on some other atlas(es)

**Comment by illdopejake** [comment]
Hey @darya-chyzhyk !

So, I was modeling some other atlases after the approach you did here. However, @kchawla-pi pointed out that I can't add in a new parameter ("atlas_filename") as the first parameter because it will screw up the positional arguments. I mean, we could do that, but we would have to implement a pretty large-scale change across many atlas and everything that calls them, and that may be too ambitious for the moment.

We talked to @GaelVaroquaux, and I will now be focusing on just harmonizing what each atlas returns (description, labels, maps), rather than harmonizing the function signature of every atlas. I won't touch Craddock, so feel free to finish this PR at your own pace.

Does that make sense?

For reference, I'll come back later and add my future PR related to this.

**Comment by kchawla-pi** [comment]
@darya-chyzhyk wrote:

I'm trying to finish my craddock_atlas PR to Nilearn and stuck with the test,

I have several error that cant debug mainly because cant run tests.
What is the procedure of running the tests on my computer?

```python
cd /nilearn/nilearn
%run datasets/tests/test_atlas
```
and got this error
```python
ImportError: cannot import name 'test_utils'
> /home/darya/Documents/github/nilearn/nilearn/datasets/tests/test_atlas.py(23)<module>()
     21 
     22 from nilearn._utils.testing import assert_raises_regex
---> 23 from . import test_utils as tst
     24 
     25 from nilearn._utils.compat import _basestring, _urllib
```
if i manually import it, I still got the error that can't really understand

1 test_downloader()
```python
<ipython-input-12-89dc78955baf> in test_downloader()
     95     local_url = "file:" + _urllib.request.pathname2url(
     96         os.path.join(tst.datadir, "craddock_2011_parcellations.tar.gz"))
---> 97     datasetdir = os.path.join(tst.tmpdir, 'craddock_2012')
     98     os.makedirs(datasetdir)
     99 

/home/darya/anaconda3/lib/python3.6/posixpath.py in join(a, *p)
     76     will be discarded.  An empty last part will result in a path that
     77     ends with a separator."""
---> 78     a = os.fspath(a)
     79     sep = _get_sep(a)
     80     path = a

TypeError: expected str, bytes or os.PathLike object, not NoneType
> /home/darya/anaconda3/lib/python3.6/posixpath.py(78)join()
     76     will be discarded.  An empty last part will result in a path that
     77     ends with a separator."""
---> 78     a = os.fspath(a)
     79     sep = _get_sep(a)
     80     path = a
```

for some reason `tst.tmpdir` is `None`.

I'm not sure how to do the test correctly. If you can guid me a little bit would be a huge help for me. I'm stuck for 1 day already

**Comment by kchawla-pi** [comment]
@darya-chyzhyk 
To run the test in the shell/bash, use one of these commands:
in the `nilearn/nilearn/tests` directory:
```shell
nose
nose <path to test file>
nose <path to test file>:<name of specific test
```

**Comment by kchawla-pi** [comment]
If you are having import troubles, navigate to nilearn project's root dir, the one with `setup.py` and use the command `pip install -e .` to install the source into your current virtual environment in editable mode.

**Comment by kchawla-pi** [comment]
@darya-chyzhyk Another way to solve import problems is to replace relative imports with absolute imports. PEP8 recommendations also promote absolute imports.

**Comment by kchawla-pi** [comment]
:wave:

**Comment by bthirion** [comment]
Stalled. Closing for now.

**Pull request closed without merging**